pkg/pillar/cas/cas.go

```diff
@@ -0,0 +1,66 @@
+package cas
+
+import (
+	"fmt"
+	"io"
+	"time"
+)
+
+//CAS: provides methods to interact with CAS clients
+type CAS interface {
+	//Blob APIs
+	//CheckBlobExists: returns true if the blob exists.
+	CheckBlobExists(blobSha256 string) bool
+	//GetBlobSize: returns size of the blob
+	GetBlobSize(blobSha256 string) (int64, error)
+	//ListBlobs: returns list of blob sha256
+	ListBlobs() ([]string, error)
+	//IngestBlob: ingests a blob from the given reader and assigns the given sha256 to it.
+	IngestBlob(blobSha256 string, reader io.Reader) error
+	//ReadBlob: returns a reader of blob
+	ReadBlob(sha256 string) (io.Reader, error)
+	//RemoveBlob: removes the blob after the given timestamp.
+	RemoveBlob(blobSha256 string, timeStamp time.Time) (*time.Timer, error)
+	//ChildHashesOfBlob: returns list of child blob sha256 if the given blobSha256 is of an Index or a Manifest.
+	//Else returns empty list.
+	ChildHashesOfBlob(blobSha256 string) ([]string, error)
+
+	//Image APIs
+	//LoadImage: creates an image from the given blobSha256. Expects the blobSha256 to be on an index
+	LoadImage(imageReference, sha256 string) error
+	//GetImageHash: returns blobSha256 (of index blob) for the given image reference
+	GetImageHash(imageReference string) (string, error)
+	//ListImages: returns a list of image references
+	ListImages() ([]string, error)
+	//RemoveImage: removes an image with the give imageReference
+	RemoveImage(imageReference string) error
+	//ReplaceImage: replaces the blobSha256 (index blob) to which the imageReference is pointing to with the given blobSha256
+	ReplaceImage(imageReference, sha256 string) error
```

Comment:
Same as above: 

* replace `imageReference` with the simpler `ref`
* make the blob sha arg name consistent
* name the blob sha arg name something non-sha256 specific
* make the format clear in the comment `<algo>:<hash>`

Reply:
Done.